addons/point_of_sale/models/pos_session.py

```diff
@@ -988,8 +980,7 @@ def _accumulate_amounts(self, data):
                         },
                         order.date_order,
                     )
-                    if self.config_id.is_closing_entry_by_product:
-                        sales[sale_key] = self._update_quantities(sales[sale_key], base_line['quantity'])
+                    sales[sale_key] = self._update_quantities(sales[sale_key], base_line['quantity'])
```

Comment:
```py
if self.config_id._is_quantities_set()
    sales[sale_key].setdefault('quantity', 0)
    sales[sale_key]['quantity'] = base_line['quantity']
```

_is_quantities_set in point_of_sale
```py
def _is_quantities_set(self):
    return self.config_id.is_closing_entry_by_product
```

_is_quantities_set in l10n_in_pos
```py
def _is_quantities_set(self):
    return self.config_id.company_id._l10n_in_is_gst_registered() or super()._is_quantities_set(self)
```

Reply:
Using this `_update_quantities` and `_update_quantity_vals` is removed and improve readability